src/NodaTime/CalendarSystem.cs

```diff
@@ -11,6 +11,11 @@
 using System.Diagnostics.CodeAnalysis;
 using static System.FormattableString;
 
+// Remove static constructors.
+// The calendar system initialization here can be slow or increase memory usage,
+// so the static constructors are very deliberately present to make them truly lazy.
```

Comment:
Didn't we have an initialisation-order issue around this area that required them to be lazy? (or was that something else?)